servicediscovery/mdns/mdns.go

```diff
@@ -42,6 +46,8 @@ func lookupAddressMDNS(id string) (string, error) {
 	var addr string
 	entries := make(chan *zeroconf.ServiceEntry)
 
+	ctx, cancel := context.WithTimeout(context.Background(), browseTimeout)
```

Comment:
can we leave the `defer cancel()` as such? From docs it seems that multiple calls to the function after the first call does nothing. 
https://golang.org/src/context/context.go?s=7999:8021#L214

Reply:
ok for the safety.